src/main/scala/Ch3.scala

```diff
@@ -125,3 +125,65 @@ object Ch3Exercises {
     g: A => C contradicts with A => B, therefore not well typed
     */
 }
+
+object Ch3Ex2 {
+
+  // Problem 1
+  sealed trait CellState {
+    def isBomb: Boolean = this match {
+      case BombCell() => true
+      case _ => false
+    }
+  }
+
+  final case class ClosedCell() extends CellState
+  final case class BombCell() extends CellState
+  final case class OpenCell(neighborBombs: Int) extends CellState
+
+  // Problem 2
+  def numCellsShowingZeroNeighborBombs(cells: Seq[Seq[CellState]]): Int = {
+    cells.flatten.filter{
```

Comment:
Instead of .filter().length, use .count()

Reply:
oh nice! thanks!